src/uct/ib/base/ib_iface.c

```diff
@@ -602,8 +623,21 @@ ucs_status_t uct_ib_iface_query(uct_ib_iface_t *iface, size_t xport_hdr_len,
     /* Calculate packet overhead  */
     mtu                   = ucs_min(uct_ib_mtu_value(active_mtu),
                                     iface->config.seg_size);
-    extra_pkt_len         = UCT_IB_LRH_LEN + UCT_IB_BTH_LEN + xport_hdr_len +
-                            UCT_IB_ICRC_LEN + UCT_IB_VCRC_LEN + UCT_IB_DELIM_LEN;
+
+#if HAVE_DECL_IBV_LINK_LAYER_INFINIBAND
+    if (uct_ib_iface_port_attr(iface)->link_layer != IBV_LINK_LAYER_ETHERNET) {
+        extra_pkt_len = UCT_IB_LRH_LEN + UCT_IB_BTH_LEN + xport_hdr_len +
+                        UCT_IB_ICRC_LEN + UCT_IB_VCRC_LEN + UCT_IB_DELIM_LEN;
+    } else {
+        extra_pkt_len = UCT_IB_GRH_LEN + UCT_IB_ROCE_LEN + UCT_IB_BTH_LEN + xport_hdr_len +
```

Comment:
also - pls try to avoid code duplication here. For example, only UCT_IB_LRH_LEN or UCT_IB_GRH_LEN + UCT_IB_ROCE_LEN should be inside the conditionals. the rest should be outside the 'if'